internal/store/chain.go

```diff
@@ -0,0 +1,203 @@
+package store
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"sync/atomic"
+
+	"github.com/eigerco/strawberry/internal/block"
+	"github.com/eigerco/strawberry/internal/crypto"
+	"github.com/eigerco/strawberry/pkg/db"
+	"github.com/eigerco/strawberry/pkg/db/pebble"
+)
+
+var (
+	ErrBlockNotFound = errors.New("block not found")
+	ErrChainClosed   = errors.New("chain store is closed")
+)
+
+const (
+	prefixHeader byte = iota + 1
+	prefixBlock
+)
+
+// Chain manages blockchain storage using a key-value store
+type Chain struct {
+	db     db.KVStore
+	closed atomic.Bool
+}
+
+// NewChain creates a new chain store using KVStore
+func NewChain(db db.KVStore) *Chain {
+	return &Chain{db: db}
+}
+
+// PutBlock stores a block and its header atomically
+func (c *Chain) PutBlock(b block.Block) error {
+	if c.closed.Load() {
+		return ErrChainClosed
+	}
+
+	// Create new batch for atomic operations
+	batch := c.db.NewBatch()
+	defer batch.Close()
+
+	headerHash, err := b.Header.Hash()
+	if err != nil {
+		return fmt.Errorf("hash header: %w", err)
+	}
+
+	// TODO: We should probably store the header here and refactor the Header file storage (AncestorStore)
```

Comment:
Maybe we could start adding issue ID when we add a TODO. This way we don't forget to add an issue that addresses the TODO. Even if it's a placeholder and it's never done, could be useful.